modules/simple-payments/simple-payments.php

```diff
@@ -0,0 +1,254 @@
+		if ( ! self::$instance ) {
+			self::$instance = new self();
+			self::$instance->register_init_hook();
+		}
+		return self::$instance;
+	}
+
+	private function register_scripts() {
+		/**
+		 * Paypal heavily discourages putting that script in your own server:
+		 * @see https://developer.paypal.com/docs/integration/direct/express-checkout/integration-jsv4/add-paypal-button/
+		 */
+		wp_register_script( 'paypal-checkout-js', 'https://www.paypalobjects.com/api/checkout.js', array(), null, true );
+		wp_register_script( 'paypal-express-checkout', plugins_url( '/paypal-express-checkout.js', __FILE__ ) , array( 'jquery', 'paypal-checkout-js' ), '0.21' );
+		wp_enqueue_style( 'simple-payments', plugins_url( '/simple-payments.css', __FILE__ ) );
+	}
+	private function register_init_hook() {
+		add_action( 'init', array( $this, 'init_hook_action' ) );
+	}
+	private function register_shortcode() {
+		add_shortcode( self::$shortcode, array( $this, 'parse_shortcode' ) );
+	}
+
+	public function init_hook_action() {
+		add_filter( 'rest_api_allowed_post_types', array( $this, 'allow_rest_api_types' ) );
+		add_filter( 'jetpack_sync_post_meta_whitelist', array( $this, 'allow_sync_post_meta' ) );
+		$this->register_scripts();
+		$this->register_shortcode();
+		$this->setup_cpts();
+
+		add_filter( 'the_content', array( $this, 'remove_auto_paragraph_from_product_description' ), 0 );
+	}
+
+	function remove_auto_paragraph_from_product_description( $content ) {
+		if ( get_post_type() === self::$post_type_product ) {
+			remove_filter( 'the_content', 'wpautop' );
+		}
+
+		return $content;
+	}
+
+	function parse_shortcode( $attrs, $content = false ) {
+		if ( empty( $attrs['id'] ) ) {
+			return;
+		}
+		$product = get_post( $attrs['id'] );
+		if ( ! $product || is_wp_error( $product ) ) {
+			return;
+		}
+		if ( $product->post_type !== self::$post_type_product ) {
+			return;
+		}
+
+		// We allow for overriding the presentation labels
+		$data = shortcode_atts( array(
+			'blog_id'     => Jetpack_Options::get_option( 'id' ),
+			'dom_id'      => uniqid( self::$css_classname_prefix . '-' . $product->ID . '_', true ),
+			'class'       => self::$css_classname_prefix . '-' . $product->ID,
+			'title'       => get_the_title( $product ),
+			'description' => $product->post_content,
+			'cta'         => get_post_meta( $product->ID, 'spay_cta', true ),
+			'multiple'    => get_post_meta( $product->ID, 'spay_multiple', true ) || '0'
+		), $attrs );
+		$data['price'] = $this->format_price(
+			get_post_meta( $product->ID, 'spay_price', true ),
+			get_post_meta( $product->ID, 'spay_currency', true ),
+			$data
+		);
+		$data['id'] = $attrs['id'];
+		if ( ! wp_script_is( 'paypal-express-checkout','enqueued' ) ) {
+			wp_enqueue_script( 'paypal-express-checkout' );
+		}
+
+		wp_add_inline_script( 'paypal-express-checkout', "try{PaypalExpressCheckout.renderButton( '{$data['blog_id']}', '{$attrs['id']}', '{$data['dom_id']}', '{$data['multiple']}' );}catch(e){}" );
```

Comment:
Paranoid mode: do you think it would be better to use `sprintf` and `esc_js` for all variables here?